### ISSUE TITLE
ReadOnly Constraint not enforced on OrientElements

### ISSUE THREAD

**Issue by shooit** [orientechnologies/orientdb#7144]
## Expected behavior  
ReadOnly properties should throw an exception if you try to update them

## Actual behavior  
No exception is thrown and the update is committed to the graph

## Steps to reproduce  
```
@Test(expected=OValidationException.class)
public void testPropertyReadOnly() {
    String location = "memory:test_readOnly";
    OrientGraphNoTx graphNoTx = new OrientGraphNoTx(location);
    OrientVertexType testType = graphNoTx.createVertexType("Test");
    OProperty prop = testType.createProperty("name", OType.STRING).setReadonly(true);
    graphNoTx.shutdown();

    Assert.assertTrue(prop.isReadonly()); //this one passes

    OrientGraph graph = new OrientGraph(location);
    try {
        OrientVertex vert1 = graph.addVertex("class:Test", "name", "Sam");
        graph.commit();

        vert1.setProperty("name", "Ben"); //should throw an exception
        graph.commit();

        Assert.assertEquals(vert1.getProperty("name"), "Sam");  //fails
    } finally {
        graph.drop();
    }
}
```

**Comment by tglman** [comment]
hi @shooit,

Reproduced this and fixed, added the relative test case, it will be released in 2.2.17.

Regards

**Comment by shooit** [comment]
@tglman looks good to me. thanks!
